signer/src/network/libp2p/bootstrap.rs

```diff
@@ -0,0 +1,447 @@
+//! LibP2P behavior for bootstrapping the node against the network using
+//! the known seed addresses.
+
+use std::{
+    collections::{HashMap, HashSet, VecDeque},
+    task::Poll,
+    time::{Duration, Instant},
+};
+
+use libp2p::{
+    core::{transport::PortUse, Endpoint},
+    swarm::{
+        dial_opts::DialOpts, dummy, ConnectionDenied, ConnectionId, FromSwarm, NetworkBehaviour,
+        THandler, THandlerInEvent, ToSwarm,
+    },
+    Multiaddr, PeerId,
+};
+
+use super::MultiaddrExt;
+
+#[derive(Clone, Debug)]
+pub struct Config {
+    local_peer_id: PeerId,
+    seed_addresses: Vec<Multiaddr>,
+    bootstrap_interval: Duration,
+    initial_delay: Duration,
+}
+
+impl Config {
+    /// Creates a new [`Config`] instance with the provided local peer ID.
+    pub fn new(local_peer_id: PeerId) -> Self {
+        Self {
+            local_peer_id,
+            seed_addresses: Default::default(),
+            bootstrap_interval: Duration::from_secs(60),
+            initial_delay: Duration::ZERO,
+        }
+    }
+
+    /// Adds seed addresses to the configuration.
+    #[allow(dead_code)]
```

Comment:
nit: we should be able to remove a couple of those `#[allow(dead_code)]` in here